src/scripts/chat-message-response.js

```diff
@@ -22,11 +22,20 @@ async function triggerResponse(actor, request) {
     let alias = await actor.getFlag("unkenny", "alias");
     request = replaceAlias(request, alias, name);
 
+    let preamble = await actor.getFlag("unkenny", "preamble") || "";
     let parameters = await getGenerationParameters(actor);
     if (!parameters) {
         return;
     }
 
+    if (await actor.getFlag("unkenny", "includeBiography") && parameters.biography.trim()) {
+        preamble += `\n\n${parameters.biography}`;
+    }
+
+    console.log("Constructed Preamble:", preamble);
+
+    request = `${preamble}\n\n${request}`;
```

Comment:
This way the preamble is included twice:
`generateResponse` calls `collectChatMessages`, which calls `messagesOrganisedForTemplate`, which prepends the preamble.